documentation/library_guide/dynamic_selection_api/round_robin_policy.rst

```diff
@@ -0,0 +1,215 @@
+and ``submit_and_wait``. It also can used with ``policy_traits``.
+
+Example
+-------
+
+The following example demonstrates a simple approach to send work to each 
+queue in a set of queues, and then wait for all devices to complete the work
+before repeating the process. A ``round_robin_policy`` is used rotate through
+the available devices.
+
+.. code:: cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+  #include <sycl/sycl.hpp>
+  #include <iostream>
+
+  const size_t N = 10000;
+  namespace ex = oneapi::dpl::experimental;
+
+  void f(sycl::handler& h, float *v);
+
+  int round_robin_example(std::vector<sycl::queue>& similar_devices, 
+                          std::vector<float *>& usm_data) {
+
+    ex::round_robin_policy p{similar_devices}; // (1)
+
+    auto num_devices = p.get_resources().size();
+    auto num_arrays = usm_data.size();
+
+    // (2)
+    auto submission_group_size = (num_arrays < num_devices) ? num_arrays : num_devices;
+
+    std::cout << "Running with " << num_devices << " queues\n"
+              << "             " << num_arrays  << " usm arrays\n"
+              << "Will perform " << submission_group_size << " concurrent offloads\n";
+
+    for (int i = 0; i < 100; i+=submission_group_size) { // (3)
+      for (int j = 0; j < submission_group_size; ++j) {  // (4)
+        ex::submit(p, [&](sycl::queue q) { // (5)
+          float *data = usm_data[j];
+          return q.submit([=](sycl::handler &h) { // (6)
+            f(h, data);
+          });
+        }); 
+      }   
+      ex::wait(p.get_submission_group()); // (7)
+    }
+    return 0;
+  }
+
+The key points in this example are:
+
+#. A ``round_robin_policy`` is constructed that rotates between the CPU and GPU queues.
+#. The total number of concurrent offloads, ``submission_group_size``, will be limited to the number of USM arrays or the number of queues, whichever is smaller. 
+#. The outer ``i``-loop iterates from 0 to 99, stepping by the ``submission_group_size``. This number of submissions will be offload concurrently.
+#. The inner ``j``-loop iterates over ``submission_group_size`` submissions.
+#. ``submit`` is used to select a queue and pass it to the user's function, but does not block until the event returned by that function completes. This provides the opportunity for concurrency across the submissions.
+#. The queue is used in function to perform an asynchronous offload. The SYCL event returned from the call to ``submit`` is returned. Returning an event is required for functions passed to ``submit`` and ``submit_and_wait``.
+#. ``wait`` is called to block for all of the concurrent ``submission_group_size`` submissions to complete.
+
+selection algorithm
+-------------------
+ 
+The selection algorithm for ``round_robin_policy`` rotates through
+the elements of the set of available resources. A simplified, expository 
+implementaton of the selection algorithm follows:

```

Comment:
Please correct spelling to "implementation"

Reply:
```suggestion
implementation of the selection algorithm follows:
```